#! /bin/sh /usr/share/dpatch/dpatch-run
## 99-unnamed.dpatch by  <blair@ctrlpclnx.nscl.msu.edu>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix up Makefile.am to shuffle the bits to match Tcl/Tk packages in Lenny

@DPATCH@

--- epicstcl-2.0-001-old/Makefile.am	2009-03-17 11:06:48.000000000 -0400
+++ epicstcl-2.0-001-new/Makefile.am	2009-03-17 14:03:57.000000000 -0400
@@ -77,7 +77,8 @@
 
 ####### The Tcl Widgets.
 
-PKGDIR=$(prefix)/TclLibs/widgets
+WIDGETSDIR=$(DESTDIR)$(prefix)/share/tcltk/epicstcl2.0
+LIBSDIR=$(DESTDIR)$(prefix)/lib/tcltk/epicstcl2.0
 
 PACKAGES = meter.tcl bcmMeter.tcl epicsLabel.tcl \
 	radioMatrix.tcl epicsEnumeratedControl.tcl \
@@ -139,20 +140,22 @@
 #### installation time stuff.
 
 install-exec-local: $(MANPAGES)
-	$(mkinstalldirs) $(PKGDIR) 
-	chmod o+rx $(PKGDIR)
-	$(INSTALL_SCRIPT) $(PACKAGES) $(PKGDIR)
-	echo "pkg_mkIndex $(PKGDIR) *.tcl" >makeindex.tcl
+	$(mkinstalldirs) $(WIDGETSDIR) 
+	chmod o+rx $(WIDGETSDIR)
+	$(INSTALL_SCRIPT) $(PACKAGES) $(WIDGETSDIR)
+	echo "pkg_mkIndex $(WIDGETSDIR) $(LIBSDIR) *.tcl" >makeindex.tcl
 	$(TCLSH_CMD) <makeindex.tcl
 
 
 #  This is to get the epics library put in the right place after install:
 
 install-data-local:
-	$(mkinstalldirs) $(prefix)/TclLibs
-	-cp $(prefix)/lib/libepics* $(prefix)/TclLibs
+	$(mkinstalldirs) $(LIBSDIR)
+	mv $(DESTDIR)$(prefix)/lib/libepics* $(DESTDIR)$(prefix)/lib/libException* $(DESTDIR)$(prefix)/lib/libtclPlus* $(LIBSDIR)
 	echo "package ifneeded epics 2.0 [list load [file join \$$dir libepics$(SLIBEXT)]]"  \
-		>> $(prefix)/TclLibs/pkgIndex.tcl;
+		>> $(LIBSDIR)/pkgIndex.tcl;
+	echo "package ifneeded epics 2.0 [list load [file join \$$dir $(prefix)/lib/tcltk/epicstcl2.0/libepics$(SLIBEXT)]]" \
+		>> $(WIDGETSDIR)/pkgIndex.tcl;
 
